Frontend/feature/transfer history UI

weilbith opened the pull request:
Closes #367 
Based on top of #683. I'll rebase as soon as that PR got merged.

Simply try it out and have fun.

nekhaly commented:
While testing, a question has just jumped to my mind, do we need to have a different history instance for each account we are connected to :thinking: ?

nekhaly reviewed (changes requested):
Except for the nitpicky comments, nothing would prevent merging this! Another piece of organized code with lovely separation and single responsibility principle following!
Thank you for the great implementation and integration of this feature! once answering the questions, feel free to ping me back so I hit the approve button gladly!

weilbith commented on frontend/src/composables/useTransferGrouping.ts:
```diff
@@ -0,0 +1,68 @@
+  transfers: Ref<Array<Transfer>>,
+  timeWindows: Ref<Array<TimeWindow>>,
+) {
+  const now = Date.now();
```
Could you provide an example how `moment` would help us here? I'm not if it is helpful at all and if, maybe just 2-3 lines or so. I'm not a big fan of buying into dependencies for such tiny things. Usually it is more work, worse testing, larger bundle to use them in such scenarios.
But maybe you have a really good idea how to use `moment` here which removes most of the code.

nekhaly replied:
OK, sure... The whole idea is about the separation of date manipulation from the component that uses the manipulated form of a date, then the decision of using any lib for this comes later!  
The questions about `date` values in the app maybe:

1- What if I need to reuse the Date manipulation/formatting in another component in the future! For example, you need the values of today, yesterday, ...etc to be listed in the filter select box which you want to add as a separate component! or maybe easier, if you want to show the last transaction date in another place other than `TransferGrouping`! Reusability questions have no end indeed!

2- The usual question about `Localization` of date! Things are easy right now, but it never keeps easy :D     
  
3- This isn't about Date values, but more about the bundler! I didn't go deep inside bundlers' internals, but aren't they picking from the 3rd-party libs (plus the core part) only the chunks of code that are imported into the app?   
  
Anyway, I'm also not a fan of using unnecessary libs, but from my experience, once the date comes into the game, the hell is looking from behind waiting for the right moment to strike (maybe that's why they call the lib `moment` :rofl: )

nekhaly replied:
OK, sure... The whole idea is about the separation of date manipulation from the component that uses the manipulated form of a date, then the decision of using any lib for this comes later!  
The questions about `date` values in the app maybe:

1- What if I need to reuse the Date manipulation/formatting in another component in the future! For example, you need the values of today, yesterday, ...etc to be listed in the filter select box which you want to add as a separate component! or maybe easier, if you want to show the last transaction date in another place other than `TransferGrouping`! Reusability questions have no end indeed!

2- The usual question about `Localization` of date! Things are easy right now, but it never keeps easy :D     
  
3- This isn't about Date values, but more about the bundler! I didn't go deep inside bundlers' internals, but aren't they picking from the 3rd-party libs (plus the core part) only the chunks of code that are imported into the app?   
  
Anyway, I'm also not a fan of using unnecessary libs, but from my experience, once the date comes into the game, the hell is looking from behind waiting for the right moment to strike (maybe that's why they call the lib `moment` :rofl: )

weilbith commented on frontend/src/actions/transfers/transfer.ts:
```diff
@@ -19,7 +19,7 @@ import type { RequestInformationData } from './request-information';
 
 const STEPS_DATA = [
-  { identifier: 'sendRequestTransaction', label: 'Please confirm your request on Metamask' },
+  { identifier: 'sendRequestTransaction', label: 'Please confirm the transaction' },
```
Okay interesting. I can clarify this again. 👍🏾 
Btw: this got shorten to make it fit without line break in the new UI.

weilbith replied:
Alright. I talked with Olympe about it.  And especially because of having WalletConnect now and the user should be informed to pick up his phone and confirm, she really likes it. She was unsure when this conversation was anyway. 😆 

Thank you for ensuring that we have the right concept in place here!

nekhaly replied:
> Btw: this got shorten to make it fit without line break in the new UI.
Oh, I thought it was a proactive action for the coming WC integration :D



> she really likes it. She was unsure when this conversation was anyway. 😆
That's normal, maybe I was imagining this as well! 🤣

weilbith commented on frontend/src/components/RequestDialog.vue:
```diff
@@ -8,33 +8,17 @@
           :disabled="!valid"
           @click="submitForm"
         >
           Transfer funds
```
I can as the UI/UX expert here. Is a bit of extra work to do that is not too nice. But could be done. I'll clarify this. 👍🏾

weilbith replied:
✔️

weilbith commented:
Alright. The last requested feature of the "New Transfer" button actually caused a lot of additional work. Not optimal and possibly could have been split. But that's how it turned out in the end. Sorry.

I'll resolve the conflicts now.

weilbith commented:
This is far from optimal, but we are currently do not have enough personnel. This feature is already delayed. As I got already a review for the majority of the commits and I addressed the feedback, I'll make use of my administrator permissions and merge the PR anyway. For future work we need to find a way to deal with PR reviews and having not enough developers.
